ntrans_gui.py

```diff
@@ -0,0 +1,261 @@
+
+        ngram_data_size = ttk.Combobox(
+            mainframe, state="readonly", textvariable=self.data_size_var
+        )
+
+        ngram_data_size["values"] = (
+            100,
+            300,
+            500,
+            800,
+            1000,
+            1500,
+            3000,
+            5000,
+            10000,
+        )
+
+        ngram_data_size.current(6)
+
+        # Generate Button
+        generate_button_style = ttk.Style()
+        generate_button_style.configure(
+            "W.TButton",
+            font=("TkDefaultFont", 16),
+            foreground="yellow",
+            background="black",
+            padding=30,
+        )
+        generate_dictionary = ttk.Button(
+            mainframe,
+            command=self.generate_ntrans_dictionary,
+            text="Generate N-Trans Dictionary",
+            style="W.TButton",
+        )
+
+        # About / Help
+        about_help_buttonframe = ttk.Frame(mainframe)
+
+        about_button = ttk.Button(
+            about_help_buttonframe,
+            command=self.open_about_window,
+            text="About",
+            style="",
+        )  # TODO: Make command into lambda instantiating about-class
+        help_button = ttk.Button(
+            about_help_buttonframe, command=self.open_help_page, text="Help"
+        )
+
+        # Black turn off formatting
+        # fmt: off
+
+        # ==================
+        # Widget placement
+        # ==================
+        header.grid(column=0, row=0, columnspan=2, padx=(0, 0), pady=(30, 30))
+
+        ngram_check_label.grid(sticky="N", column=0, row=1, padx=(20, 0), pady=(10, 10))
+
+        select_all_checkbutton.grid(sticky="W", column=0, row=2, padx=(40, 0), pady=(0, 10))
+
+        for n in n_to_ngram_checkbox:
+            n_to_ngram_checkbox[n].grid(sticky="W", column=0, padx=(40, 0), pady=(0, 0))
+
+        data_size_label.grid(sticky="W", column=0, row=8, padx=(20, 0), pady=(30, 0))
+        ngram_data_size.grid(sticky="W", column=0, row=9, padx=(20, 0), pady=(5, 0))
+
+        target_language_label.grid(sticky="W", column=0, row=10, padx=(20, 20), pady=(30, 0))
+        target_language.grid(sticky="W", column=0, row=11, padx=(20, 20), pady=(0, 0))
+
+        filepath_label.grid(sticky="W", column=0, row=12, padx=(20, 0), pady=(30, 10))
+        show_filepath.grid(sticky="W", column=0, row=13, padx=(20, 0), pady=(0, 10))
+        get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 10))
+
+        generate_dictionary.grid(column=0, row=14, columnspan=2, padx=(0, 0), pady=(30, 30))
+
+        about_help_buttonframe.grid(sticky="E", column=0, row=15, columnspan=2, padx=(20, 20), pady=(0, 10))
+        about_button.pack(side="left")
+        help_button.pack(side="right")
+
+        # Black turn on formatting
+        # fmt: on
+
+    def get_save_file_path(self) -> None:
+        self.savepath: str = filedialog.askdirectory()  # type: ignore
```

Comment:
Needed to ignore because "error: Call to untyped function "askdirectory" in typed context".

To investigate the issue and try to solve it myself, I 
1) looked for tkinter stubs (but I don't exactly understand stubs so no luck)
2) Cloned into porcupine, git greped filedialog and tried to run mypy one some of the files. You seem to get the same error in certain files. I found one instance where you had defined "variable: str = filedialog.......". This didn't work for me however, even though I added print(type(self.savepath)), which gave "str".

Reply:
Make issue on typeshed so I remember to fix it

Reply:
Also, no need to put filepath to self. It can be a local variable.

Reply:
https://github.com/python/typeshed/issues/5571